### ISSUE TITLE
Could not get unknown property 'errorprone' for object of type org.gradle.api.tasks.compile.CompileOptions.

### ISSUE THREAD

**Issue by boris-petrov** [tbroyer/gradle-errorprone-plugin#3]
I've been using the other plugin until now (`net.ltgt.errorprone`) but I decided to give this one a try with JDK9. I only change `id 'net.ltgt.errorprone' version '0.0.14'` to `id 'net.ltgt.errorprone-javacplugin' version '0.2'` and add `options.errorprone.disableWarningsInGeneratedCode = true` in my `tasks.withType(JavaCompile)` and I get the error from the title:

```
Could not get unknown property 'errorprone' for object of type org.gradle.api.tasks.compile.CompileOptions.
```

I'm not sure what I am doing wrong. Any ideas?

P.S. JDK 9.0.4 and Gradle 4.7.

**Comment by boris-petrov** [comment]
I thought it could be because of another plugin. And yes, it is. Adding `id 'me.champeau.gradle.jmh' version '0.4.5'` breaks it. Any ideas why and where the problem might be?

**Comment by tbroyer** [comment]
`me.champeau.gradle.jmh` plugin creates a `jmhCompileGeneratedClasses` `JavaCompile` task that's not "linked" to any source set, so the extension is not automatically applied to it, as its `options.annotationProcessorPath` is not guaranteed to be set to contain Error Prone (and indeed it's left empty/null). You need to explicitly enable Error Prone on that task if you want it, and set its `options.annotationProcessorPath`. Something like:
```gradle
jmhCompileGeneratedClasses {
  ErrorProneJavacPlugin.apply(options)
  options.annotationProcessorPath = configurations.errorprone
}
```
Then you'll be able to use `options.errorprone.…` in your `tasks.withType(JavaCompile)`.

See the note about “custom `JavaCompile` tasks” in the README.

**Comment by tbroyer** [comment]
Fwiw, I just published v0.4 that applies the `options.errorprone` extension to all `JavaCompile` tasks, so `tasks.withType(JavaCompile) { options.errorprone { … } }` will no longer fail, even in the presence of the JMH plugin. For tasks that are not "linked" to a source set, `options.errorprone.enabled` defaults to `false`, so the JMH task won't use Error Prone. If you want it to use Error Prone too, then you can do:
```gradle
jmhCompileGeneratedClasses {
  options.annotationProcessorPath = configurations.errorprone
  options.errorprone.enabled = true
}
```
README has been updated.

(also please note that `ErrorProneJavacPlugin.apply(options)` API has been removed)
